articles/building-a-nodejs-application-using-docker/index.md

```diff
@@ -0,0 +1,241 @@
+# Building A Node.js Application Using Docker
+
+[Docker](https://www.docker.com/) allows us to run our applications as *containers*. A container is a standalone executable package which is lightweight and has everything needed to run an application be it libraries, tools, runtime, setting or code.
+
+In this tutorial, we will build a Node.js application, create an image and also build a container using the image. Enjoy!!!
+
+
+# Dockerize the Node.js Application
+
+To start this tutorial, we need to make sure Docker and [Node.js](https://nodejs.org/en/download/package-manager/) is installed on our system. If Docker is not, you can refer to the links below to download:
+- [Ubuntu](https://docs.docker.com/engine/installation/linux/ubuntu/)
+- [Windows](https://docs.docker.com/docker-for-windows/install/)
+- [MacOS](https://docs.docker.com/docker-for-mac/install/)
+
+In order to dockerize a Node.js Application we need to follow these steps:
+
+- Creating A Node.js Application
+- Create A DockerFile
+- Building your Docker Image
+- Expose
+
+# Step 1 - Creating a Node.js Application
+
+First, we will start by creating a directory for our project then install some dependencies for our simple Hello World Website.
+```
+ mkdir node-web-app
+ cd node-web-app
+```
+
+### Install NPM and Express Framework
+
+Install npm and install the express framework, which is one of Node.js frameworks. Initialize npm in our directory.
+```
+ npm init
+```
+NPM creates a `package.json` that holds the dependencies of the app. Install the express framework dependency.
+```
+ npm install express --save
+```
+Result:
+```
+{
+  "name": "web-app",
+  "version": "1.0.0",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
+    "express": "^4.17.1"
+  },
+  "devDependencies": {},
+  "description": ""
+}
+
+```
+Create `app.js` file with an http server that will set up the Hello World site:
+```
+//Load express module with `require` directive
+var express = require('express')
+var app = express()
+
+//Define request response in root URL (/)
+app.get('/', function (req, res) {
+  res.send('Hello World!')
+})
+
+//Launch listening server on port 8080
+app.listen(8080, function () {
+  console.log('app listening on port 8080!')
+})
+```
+
+### Run the application
+
+The app is ready to launch:
+```
+$ node app.js
+```
+Go to `http://localhost:8080/` in your browser to view it
+
+# Step 2 - Create A DockerFile
+
+Create a file in the directory called `Dockerfile`
+
+The first thing is to define which image we want to build from. Here we will use version `9` of `node` available from Docker Hub:
+```
+FROM node:9
+```
+Next is to create the working directory for your application
```

Comment:
Next should have a comma after it. Would flow better with Next, create a working directory